Adapters/Vungle/ISVungleAdapter/ISVungleBannerAdapterRouter.m

```diff
@@ -0,0 +1,212 @@
+        if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
+            vungleAdSize = BannerSizeLeaderboard;
+        } else {
+            vungleAdSize = BannerSizeRegular;
+        }
+    }
+
+    return vungleAdSize;
+}
+
+- (CGRect)getAdViewRect:(BannerSize)bannerSize {
+    CGRect rect = CGRectMake(0, 0, 320, 50);
+    switch (bannerSize) {
+        case BannerSizeShort:
+            rect = CGRectMake(0, 0, 300, 50);
+            break;
+        case BannerSizeLeaderboard:
+            rect = CGRectMake(0, 0, 728, 90);
+            break;
+        case BannerSizeMrec:
+            rect = CGRectMake(0, 0, 300, 250);
+            break;
+        default:
+            break;
+    }
+
+    return rect;
+}
+
+- (NSString *)getBannerStateString:(BANNER_STATE)state {
+    switch (state) {
+        case UNKNOWN:
+            return @"UNKNOWN";
+        case REQUESTING:
+            return @"REQUESTING";
+        case REQUESTING_RELOAD:
+            return @"REQUESTING_RELOAD";
+        case SHOWING:
+            return @"SHOWING";
+    }
+
+    return @"UNKNOWN";
+}
+
+#pragma mark - VungleBannerDelegate
+
+- (void)bannerAdDidLoad:(VungleBanner * _Nonnull)banner {
+    LogAdapterDelegate_Internal(@"placementID = %@, currentBannerState = %@", banner.placementId, [self getBannerStateString:self.bannerState]);
+
+    if (self.bannerState == REQUESTING) {
+        if (![banner canPlayAd]) {
+            // When Banner Ad is loaded the canPlayAd should also return YES
+            // If for some reason that is not the case we might want to not show the banner
+            LogAdapterDelegate_Internal(@"Vungle Banner Ad is loaded but not ready to be shown");
+
+            // update Banner state - UNKNOWN
+            self.bannerState = UNKNOWN;
+            NSError *error = [ISError createError:ERROR_BN_LOAD_NO_FILL
+                                           withMessage:[NSString stringWithFormat:@"Vungle - banner no ads to show for placementID = %@", banner.placementId]];
+            [self.delegate adapterBannerDidFailToLoadWithError:error];
+            return;
+        }
+
+        [self showBannerAd];
+        [self.delegate adapterBannerDidLoad:self.adView];
+    }
+}
+
+- (void)bannerAdDidFailToLoad:(VungleBanner * _Nonnull)banner withError:(NSError * _Nonnull)error {
+    LogAdapterDelegate_Internal(@"placementID = %@, error = %@", banner.placementId, error);
+
+    // update Banner state - UNKNOWN
+    self.bannerState = UNKNOWN;
+    NSError *smashError = [ISError createError:ERROR_BN_LOAD_NO_FILL
+                                   withMessage:[NSString stringWithFormat:@"Vungle - banner no ads to show for placementID = %@", banner.placementId]];
+    [self.delegate adapterBannerDidFailToLoadWithError:smashError];
+}
+
+- (void)bannerAdDidTrackImpression:(VungleBanner * _Nonnull)banner {
+    LogAdapterDelegate_Internal(@"placementID = %@", banner.placementId);
+    [self.delegate adapterBannerDidShow];
```

Comment:
The SDK has a didShow callback. Is this the correct location to call ironSource's callback?

Reply:
Or is this ironSource's equivalent of an impression callback?

Reply:
@stanley-vungle Hi Stanley, in 6.x version, the `adapterBannerDidShow` is called in `- (void)vungleAdViewedForPlacement:` Vungle delegate method which used to to track views. So I put `adapterBannerDidShow` in `- (void)bannerAdDidTrackImpression:` method in 7.0 version. Thanks a lot. :)